pkg/bitwarden/cli.go

```diff
@@ -142,3 +144,170 @@ func (c *cliClient) GetPassword(itemName string) ([]byte, error) {
 func (c *cliClient) Logout() ([]byte, error) {
 	return c.run("logout")
 }
+
+func createItem(itemTemplate string, targetItem *Item, c *cliClient) error {
+	// the bitwarden cli expects the item to be base64 encoded
+	encItem := b64.StdEncoding.EncodeToString([]byte(itemTemplate))
+	out, err := c.run("--session", c.session, "create", "item", encItem)
+	if err != nil {
+		return err
+	}
+	err = json.Unmarshal(out, targetItem)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func createAttachment(fileContents []byte, fileName string, itemID string, newAttachment *Attachment, c *cliClient) error {
+	filePath := "/tmp/" + fileName
+	if err := ioutil.WriteFile(filePath, fileContents, 0644); err != nil {
+		return fmt.Errorf("failed to create temporary file for new attachment: %w", err)
+	}
+	out, err := c.run("--session", c.session, "create", "attachment", "--file", filePath, "--itemid", itemID)
+	if err != nil {
+		return fmt.Errorf("bw create failed: %w", err)
+	}
+	if err = json.Unmarshal(out, newAttachment); err != nil {
+		return fmt.Errorf("failed to parse bw output: %w", err)
+	}
+
+	return nil
+}
+
+func createEmptyItem(itemName string, targetItem *Item, c *cliClient) error {
+	itemTemplate := fmt.Sprintf(`{"type":1,"name":"%s","login":{}}`, itemName)
+	return createItem(itemTemplate, targetItem, c)
+}
+
+func createItemWithPassword(itemName string, password []byte, targetItem *Item, c *cliClient) error {
+	itemTemplate := fmt.Sprintf(`{"type":1,"name":"%s","login":{"password":"%s"}}`, itemName, string(password))
+	return createItem(itemTemplate, targetItem, c)
+}
+
+func editItem(targetItem Item, c *cliClient) error {
+	targetJSON, err := json.Marshal(targetItem)
+	if err != nil {
+		return fmt.Errorf("failed to marshal object: %w", err)
+	}
+	encodedItem := b64.StdEncoding.EncodeToString(targetJSON)
+	_, err = c.run("--session", c.session, "edit", "item", targetItem.ID, encodedItem)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func deleteAttachment(attachmentID, itemID string, c *cliClient) error {
+	if _, err := c.run("--session", c.session, "delete", "attachment", attachmentID, "--itemid", itemID); err != nil {
+		return fmt.Errorf("failed to delete attachment, attachmentID: %s, itemID: %s: %w", attachmentID, itemID, err)
+	}
+	return nil
+}
+
+func (c *cliClient) SetFieldOnItem(itemName, fieldName string, fieldValue []byte) error {
+	itemFound := false
+	fieldFound := false
+	var targetItem Item
+	var targetItemIndex int
+	for index, item := range c.savedItems {
+		if itemName == item.Name {
+			itemFound = true
+			targetItem = item
+			targetItemIndex = index
+			for fieldIndex, field := range item.Fields {
+				if field.Name == fieldName {
+					fieldFound = true
+					if field.Value != string(fieldValue) {
+						if err := editItem(targetItem, c); err != nil {
+							return fmt.Errorf("failed to set field, itemName: %s, fieldName: %s - %w", itemName, fieldName, err)
+						}
```

Comment:
This can be handled outside the loop so we get a clear separation between the "search" and the "modify" phases.  We're also not `break`ing, so subsequent items are processed unnecessarily.

We can also use pointers instead of `found`/`target` pairs.

Similarly for the other functions below and the ones in `fake.go`.

Reply:
Yup I changed this as well